test/e2e/operator.go

```diff
@@ -272,3 +277,99 @@ var _ = Describe("ARO Operator - Conditions", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 })
+
+var _ = Describe("ARO Operator - Azure NSG Reconciler", func() {
+	var location, resourceGroup, subnet, vnet string
+	var correctNSG *string
+	var dummynsg mgmtnetwork.SecurityGroup
+	type subnetDescriptor struct {
+		resourceGroup string
+		vnetName      string
+		subnetName    string
+		location      string
+	}
+	BeforeEach(func() {
+		enableReconcileNSG := func() {
+			instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(context.Background(), arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			instance.Spec.Features.ReconcileNSGs = true
```

Comment:
Check if we need to enable. Once this becomes default we will not. So avoid future call :)

Reply:
good call, changed to only update instance when `!ReconcileNSGs`